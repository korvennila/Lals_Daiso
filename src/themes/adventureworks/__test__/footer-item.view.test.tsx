/*!
 * Copyright (c) Microsoft Corporation.
 * All rights reserved. See LICENSE in the project root for license information.
 */

import { buildMockModuleProps } from '@msdyn365-commerce/core';
import { IFooterItemConfig, IFooterItemProps, IFooterViewProps } from '@msdyn365-commerce-modules/footer';
import { render } from 'enzyme';
import * as React from 'react';

import FooterItemView from '../views/footer-item.view';

describe('footer item view tests', () => {
    it('renders correctly', () => {
        const moduleConfig: IFooterItemConfig = {
            className: 'class-mock'
        };

        // @ts-expect-error
        const moduleProps: IFooterItemProps<IFooterItemConfig> = buildMockModuleProps({}, {}, moduleConfig);
        const viewprops: IFooterViewProps = {
            ...moduleProps,
            heading: '{heading}',
            text: '{text}',
            image: '{image}',
            linkText: 'link text',
            FooterItem: { moduleProps, className: 'module-class-FooterItem' },
            Link: { className: 'mock-class-Link' },
            HeadingContainer: { className: 'mock-class-HeadingContainer' },
            ItemContainer: { className: 'mock-class-ItemContainer' },
            EditableLink: '{link}'
        };

        const view = render(<FooterItemView {...viewprops} />);
        expect(view).toMatchSnapshot();
    });

    it('renders correctly without heading', () => {
        const moduleConfig: IFooterItemConfig = {
            className: 'class-mock'
        };

        // @ts-expect-error
        const moduleProps: IFooterItemProps<IFooterItemConfig> = buildMockModuleProps({}, {}, moduleConfig);
        const viewprops: IFooterViewProps = {
            ...moduleProps,
            heading: '',
            text: '{text}',
            image: '{image}',
            linkText: 'link text',
            FooterItem: { moduleProps, className: 'module-class-FooterItem' },
            Link: { className: 'mock-class-Link' },
            HeadingContainer: { className: 'mock-class-HeadingContainer' },
            ItemContainer: { className: 'mock-class-ItemContainer' },
            EditableLink: '{link}'
        };

        const view = render(<FooterItemView {...viewprops} />);
        expect(view).toMatchSnapshot();
    });

    it('renders correctly without heading & image', () => {
        const moduleConfig: IFooterItemConfig = {
            className: 'class-mock'
        };

        // @ts-expect-error
        const moduleProps: IFooterItemProps<IFooterItemConfig> = buildMockModuleProps({}, {}, moduleConfig);
        const viewprops: IFooterViewProps = {
            ...moduleProps,
            heading: '',
            text: '{text}',
            image: '',
            linkText: 'link text',
            FooterItem: { moduleProps, className: 'module-class-FooterItem' },
            Link: { className: 'mock-class-Link' },
            HeadingContainer: { className: 'mock-class-HeadingContainer' },
            ItemContainer: { className: 'mock-class-ItemContainer' },
            EditableLink: '{link}'
        };

        const view = render(<FooterItemView {...viewprops} />);
        expect(view).toMatchSnapshot();
    });
});
