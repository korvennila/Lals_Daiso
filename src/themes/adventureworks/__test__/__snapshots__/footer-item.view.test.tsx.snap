// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer item view tests renders correctly 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "module-class-FooterItem",
      "data-m-t": "mock-type",
      "id": "",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "mock-class-HeadingContainer",
        },
        "children": Array [
          Node {
            "data": " ",
            "next": Node {
              "attribs": Object {
                "aria-hidden": "true",
                "class": "ms-footer__heading__title-icon",
              },
              "children": Array [],
              "name": "i",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "data": "{heading} ",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "text",
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "aria-hidden": undefined,
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "aria-hidden": undefined,
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
          Node {
            "attribs": Object {
              "aria-hidden": "true",
              "class": "ms-footer__heading__title-icon",
            },
            "children": Array [],
            "name": "i",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Node {
              "data": "{heading} ",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "text",
            },
            "parent": [Circular],
            "prev": Node {
              "data": " ",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "aria-hidden": undefined,
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "aria-hidden": undefined,
              "class": undefined,
            },
          },
          Node {
            "data": "{heading} ",
            "next": null,
            "parent": [Circular],
            "prev": Node {
              "attribs": Object {
                "aria-hidden": "true",
                "class": "ms-footer__heading__title-icon",
              },
              "children": Array [],
              "name": "i",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": Node {
                "data": " ",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "aria-hidden": undefined,
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "aria-hidden": undefined,
                "class": undefined,
              },
            },
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "data-m-t": undefined,
      "id": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "data-m-t": undefined,
      "id": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`footer item view tests renders correctly without heading & image 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "module-class-FooterItem",
      "data-m-t": "mock-type",
      "id": "",
    },
    "children": Array [
      Node {
        "data": "{link}",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "data-m-t": undefined,
      "id": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "data-m-t": undefined,
      "id": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`footer item view tests renders correctly without heading 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "module-class-FooterItem",
      "data-m-t": "mock-type",
      "id": "",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "mock-class-Link",
        },
        "children": Array [
          Node {
            "data": "link text{image}",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "data-m-t": undefined,
      "id": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "data-m-t": undefined,
      "id": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
